src/access.rs

```diff
@@ -0,0 +1,339 @@
+											merkle_proof.pos,
+										));
+										continue;
+									}
+								}
+							}
+						}
+					}
+					unconfirmed_registered_txs.push(txid);
+				}
+
+				// Check all registered outputs for dependent spending transactions.
+				let registered_outputs: Vec<WatchedOutput> = locked_watched_outputs
+					.iter()
+					.chain(locked_queued_outputs.iter())
+					.cloned()
+					.collect();
+
+				// Remember all registered outputs that haven't been spent for future processing.
+				let mut unspent_registered_outputs = Vec::new();
+
+				for output in registered_outputs {
+					if let Some(output_status) = client
+						.get_output_status(&output.outpoint.txid, output.outpoint.index as u64)?
+					{
+						if output_status.spent {
+							if let Some(spending_tx_status) = output_status.status {
+								if spending_tx_status.confirmed {
+									let spending_txid = output_status.txid.unwrap();
+									if let Some(spending_tx) = client.get_tx(&spending_txid)? {
+										let block_height = spending_tx_status.block_height.unwrap();
+										let block_header = client.get_header(block_height)?;
+										if let Some(merkle_proof) =
+											client.get_merkle_proof(&spending_txid)?
+										{
+											confirmed_txs.push((
+												spending_tx,
+												block_height,
+												block_header,
+												merkle_proof.pos,
+											));
+											continue;
+										}
+									}
+								}
+							}
+						}
+					}
+					unspent_registered_outputs.push(output);
+				}
+
+				// Sort all confirmed transactions first by block height, then by in-block
+				// position, and finally feed them to the interface in order.
+				confirmed_txs.sort_unstable_by(
+					|(_, block_height1, _, pos1), (_, block_height2, _, pos2)| {
+						block_height1.cmp(&block_height2).then_with(|| pos1.cmp(&pos2))
+					},
+				);
+				for (tx, block_height, block_header, pos) in confirmed_txs {
+					for c in &confirmables {
+						c.transactions_confirmed(&block_header, &[(pos, &tx)], block_height);
+					}
+				}
+
+				*locked_watched_transactions = unconfirmed_registered_txs;
+				*locked_queued_transactions = Vec::new();
+				*locked_watched_outputs = unspent_registered_outputs;
+				*locked_queued_outputs = Vec::new();
+			}
+
+			{
+				// Query the interface for relevant txids and check whether they have been
+				// reorged-out of the chain.
+				let unconfirmed_txids = confirmables
+					.iter()
+					.flat_map(|c| c.get_relevant_txids())
+					.filter(|txid| {
+						client
+							.get_tx_status(txid)
+							.ok()
+							.unwrap_or(None)
+							.map_or(true, |status| !status.confirmed)
+					})
+					.collect::<Vec<Txid>>();
+
+				// Mark all relevant unconfirmed transactions as unconfirmed.
+				for txid in &unconfirmed_txids {
+					for c in &confirmables {
+						c.transaction_unconfirmed(txid);
+					}
+				}
+			}
+		}
+
+		// TODO: check whether new outputs have been registered by now and process them
```

Comment:
is this required ? 
won't it just sync in next sync call ?

Reply:
imo, this shouldn't be required. 
if we called sync from outside this chain-access interface, we synced according to tx's and outputs available at that point in time. 
when we release the lock, there can be another set of tx and outputs added but that should be handled in next call. we can not make sure to sync everything at this place.